sqlglot/dialects/presto.py

```diff
@@ -169,6 +169,22 @@ def _first_last_sql(self: Presto.Generator, expression: exp.First | exp.Last) ->
     return rename_func("ARBITRARY")(self, expression)
 
 
+def _unix_to_time_sql(self: Presto.Generator, expression: exp.UnixToTime) -> str:
+    scale = expression.args.get("scale")
+    timestamp = self.sql(expression, "this")
+    if scale in {None, exp.UnixToTime.SECONDS}:
+        return rename_func("FROM_UNIXTIME")(self, expression)
+    if scale == exp.UnixToTime.MILLIS:
+        return f"FROM_UNIXTIME(CAST({timestamp} AS DOUBLE) / 1000)"
+    if scale == exp.UnixToTime.MICROS:
+        return f"FROM_UNIXTIME(CAST({timestamp} AS DOUBLE) / 1000000)"
+    if scale == exp.UnixToTime.NANOS:
```

Comment:
many nanos will lose precision if you use a double, you probably want to cast to a bigint